build.gradle

```
@@ -1 +1,14 @@
+plugins{
+    id 'com.android.application' version '8.7.2' apply false
+    id 'com.android.library' version '8.7.2' apply false
+}
 
+String getGitHash(String project) {
+    def command = Runtime.getRuntime().exec("git rev-list -1 HEAD " + project)
+    def result = command.waitFor()
+    if (result != 0) {
+        throw new IOException("Command 'getGitHash()' exited with " + result)
```

Note for a future PR/commit:

Enforce failing the build if the repository isn't cloned with Git. Not cloning with Git breaks submodules and this code already.

---

How do you suppose just letting it silently fail and not edit the versions file?